pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```diff
@@ -412,17 +423,40 @@ private void fetchResizingConfig() {
     }
   }
 
-  private void setMaxNumRecordsConfigIfSpecified(Job job) {
+  // TODO: Handle case where #records exceed long max
```

Comment:
Exceed long max? We don't need to consider Long overflow

Reply:
maybe.. for refresh use cases i coluld see it

Reply:
That is more than the number of molecules on earth lol...